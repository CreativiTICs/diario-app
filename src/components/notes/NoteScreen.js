import React from "react";
import { NotesAppBar } from "./NotesAppBar";

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
      <NotesAppBar />
      <div className="notes__content">
        <input
          type="text"
          placeholder="Algún título asombroso"
          className="notes__title-input"
          autoComplete="off"
        />
        <textarea
          placeholder="¿Qué pasó hoy?"
          className="notes__textarea"
        ></textarea>
        <div className="notes__image">
          <img
            src="https://s3-us-west-2.amazonaws.com/melingoimages/Images/54372.jpg"
            alt="Mujer pintando en Paisaje"
          />
        </div>
      </div>
    </div>
  );
};
