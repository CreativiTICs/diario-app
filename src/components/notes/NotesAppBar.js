import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startSaveNote, startUploading } from "../../actions/notes";

export const NotesAppBar = () => {
  const dispatch = useDispatch();
  const { active } = useSelector((state) => state.notes);
  const handleSave = () => {
    dispatch(startSaveNote(active));
  };

  const handlePictureUpload = () =>{
    document.querySelector("#fileSelector").click();
  }

  const handleFileChange = (e) => {
    const file = e.target.files[0];
    if(file){
      dispatch(startUploading(file));
    }
  }

  return (
    <div className="note__appbar">
      <span>28 de Diciembre 2020</span>
      <input 
      id='fileSelector'
      type="file"
      name="file"
      style={{display: 'none'}}
      onChange={handleFileChange}
      />
      <div>
        <button className="btn" onClick={handlePictureUpload} >Cargar Imagen</button>
        <button className="btn" onClick={handleSave}>
          Guardar
        </button>
      </div>
    </div>
  );
};
