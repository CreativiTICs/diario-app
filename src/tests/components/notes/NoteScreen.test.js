import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";

import { NoteScreen } from "../../../components/notes/NoteScreen";

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { activeNote } from "../../../actions/notes";

jest.mock("../../../actions/notes", () => ({
  activeNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {
    uid: "1",
    name: "Carlos",
  },
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: {
      id: 1234,
      title: "Hola",
      body: "mundo",
      date: 0,
    },
    notes: [],
  },
};
//Simular el store
let store = mockStore(initState);
//Cambiar el dispatch por una fn jest
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <NoteScreen />
  </Provider>
);

describe("Pruebas en <NoteScreen/>", () => {
  test("Debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe disparar el active note", () => {
    wrapper.find('input[name="title"]').simulate("change", {
      target: {
        name: "title",
        value: "Hola de nuevo",
      },
    });

    expect(activeNote).toHaveBeenLastCalledWith(1234, {
      body: "mundo",
      title: "Hola de nuevo",
      id: 1234,
      date: 0,
    });
  });
});
