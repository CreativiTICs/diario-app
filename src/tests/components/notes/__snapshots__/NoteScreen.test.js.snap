// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NoteScreen/> Debe mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="note__appbar"
  >
    <span>
      28 de Diciembre 2020
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Cargar Imagen
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Guardar
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Algún título asombroso"
      type="text"
      value="Hola"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="¿Qué pasó hoy?"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</div>
`;
